scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.ApplePayEventReceiver.cs.erb

```diff
@@ -29,16 +30,44 @@ namespace <%= namespace %>.SDK.iOS {
             var contentDictionary = (Dictionary<string, object>)Json.Deserialize(message.Content);
             var mailingAddressInput = new MailingAddressInput(contentDictionary);
 
-            List<SummaryItem> summaryItems = null;
-            List<ShippingMethod> shippingMethods = null;
+            Action respondWithSuccess = () => {
+                message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Success, GetSummaryItems(), GetShippingMethods()).ToString());
+            };
+
+            Action<ShopifyError> respondToError = (ShopifyError error) => {
+                switch(error.Type) {
+                case ShopifyError.ErrorType.GraphQL:
+                case ShopifyError.ErrorType.HTTP:
+                    message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure).ToString());
+                    break;
+                case ShopifyError.ErrorType.UserError:
+                    var userErrors = CurrentCart.UserErrors;
+
+                    if (userErrors.Count > 0) {
+                        ApplePayAuthorizationStatus statusToReturn = ApplePayAuthorizationStatus.Failure;
+                        foreach (var userError in userErrors) {
+                            var status = AuthorizationStatusForPreliminaryShippingUserError(userError);
+
+                            if (status == ApplePayAuthorizationStatus.Failure) {
+                                message.Respond(new ApplePayEventResponse(status).ToString());
+                                return;
+                            }
+
+                            statusToReturn = status;
+                        }
+                        message.Respond(new ApplePayEventResponse(statusToReturn, GetSummaryItems()).ToString());
+                    } else {
+                        message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure).ToString());
+                    }
+                    break;
+                }
```

Comment:
Might be good to do at the top of the method:
`ApplePayEventResponse response = null;`

Then at the bottom:
`message.Respond(response.ToString());`

Instead of having multiple:
`message.Respond`